diag_manager/fms_diag_outfield.F90

```diff
@@ -0,0 +1,212 @@
+   !!      and so may need to use the pre-processor.
+
+
+!> #brief initialize all the memebers of the class.
+   SUBROUTINE initialize_outfield_index_type(this, is, js , ks, ie, je, ke, hi, hj, f1, f2, f3, f4)
+      CLASS(fms_diag_outfield_index_type), INTENT(inout)  :: this
+      INTEGER, INTENT(in) :: is, js, ks !< Variable used to update class member of same names.
+      INTEGER, INTENT(in) :: ie, je, ke !< Variable used to update class member of same names.
+      INTEGER, INTENT(in) :: hi, hj !< Variable used to update class member of same names.
+      INTEGER, INTENT(in) :: f1, f2, f3, f4 !< Variable used to update class member of same names.
+
+      this%is = is
+      this%js = js
+      this%ks = ks
+      this%ie = ie
+      this%je = je
+      this%ke = ke
+
+      this%hi = hi
+      this%hj = hj
+
+      this%f1 = f1
+      this%f2 = f2
+      this%f3 = f3
+      this%f4 = f4
+   END SUBROUTINE initialize_outfield_index_type
+
+
+   !> @brief Update with those fields used in the legacy diag manager.
+  !! Note that this is initializing from the legacy structures.
+   !! Note that output_frequency  came from file_type;
+   SUBROUTINE initialize_outfield_imp(this, input_field,  output_field, mask_present, freq)
+      CLASS(fms_diag_outfield_type), INTENT(inout) :: this
+      TYPE(input_field_type),     INTENT(in) :: input_field
+      TYPE(output_field_type),    INTENT(in) :: output_field
+      LOGICAL, INTENT(in) :: mask_present
+      INTEGER, INTENT(in) :: freq
+      INTEGER ::  time_redux
+
+      this%module_name = input_field%module_name
+      this%field_name = input_field%field_name
+      this%output_name = output_field%output_name
+
+      this%pow_value = output_field%pow_value
+      this%phys_window = output_field%phys_window
+      this%need_compute = output_field%need_compute
+      this%reduced_k_range = output_field%reduced_k_range
+      this%mask_variant = input_field%mask_variant
+      !!Note: in legacy diag manager, presence of missing value vs presence of mask
+      !! is determined in different ways (diag table vs send function call)
+      this%missvalue_present = input_field%missing_value_present
+      this%mask_present = mask_present
+
+      time_redux = get_output_field_time_reduction (output_field)
+      call this%time_reduction%initialize( time_redux , freq)
+
+      !!TODO: the time_min and time_max buffer update code is almost the exact same src code, except
+      !!  for the compariosn function. Simplify code and set comparison function:
+      !!TODO: If possible add to the power function. See issue with pointers and elemental functions
+
+   END SUBROUTINE initialize_outfield_imp
+
+
+   !> \brief Get the time reduction from a legacy output field.
+    !! Note we do not place this in the time_reduction class to avoid circular dependencies.
+   function get_output_field_time_reduction(ofield) result (rslt)
+    TYPE(output_field_type), INTENT(in) :: ofield
```

Comment:
This variable needs to be documented in doxygen format

Reply:
done